### PR TITLE
Traversal for VerletClusterLists + Refactoring

### PR DESCRIPTION
**Pull request by HobbyProgrammer**
# Description

`iteratePairwise` of `VerletClusterLists` now properly uses a traversal. Old implementation was moved into a new traversal class.

While reading through the code, I also did some extract-method refactoring of `VerletClusterLists::rebuild()` and `VerletClusterLists::updateVerletLists()` to help future maintainers understand the code better. Some of the extracted methods have quite large argument lists, so it is not perfect, but overall, in my opinion, it is better. The refactoring can easily be reverted, if you think differently.

## Related Pull Requests

- #260 Needs this, if you want to continue with that pull request.

## Resolved Issues

- [x] #Issue #155 Traversal has been implemented

# How Has This Been Tested?

- Jenkins

### CODE REVIEW

File: tests/testAutopas/tests/selectors/AutoTunerTest.cpp [hunk 1]
```diff
@@ -33,11 +33,25 @@ TEST_F(AutoTunerTest, testAllConfigurations) {
...
+  //                      c18                 with (AoS <=> SoA, newton3 <=> noNewton3) = 4
+  //                      c01                 with (AoS <=> SoA, noNewton3)             = 2
+  //                      c01-combined-SoA    with (SoA, noNewton3)                     = 1
+  // VerletLists:         verlet-lists        with (AoS <=> SoA, newton3 <=> noNewton3) = 4
+  // VerletListsCells:    verlet-sliced       with (AoS, newton3 <=> noNewton3)         = 2
+  //                      verlet-c18          with (AoS, newton3 <=> noNewton3)         = 2
+  //                      verlet-c01          with (AoS, noNewton3)                     = 1
+  // VerletClusterLists:  verlet-clusters     with (AoS, noNewton3)                     = 1
+  //                                                                               --------
+  //                                                                                     29
```
Comment (FG-TUM) [inline tests/testAutopas/tests/selectors/AutoTunerTest.cpp]: nice representation! Including CUDA stuff would also be helpful (-> @julian-spahl )

**Comment by HobbyProgrammer** [comment]
@FG-TUM The PR is ready to merge. Jenkins has some connections problems, though, so it might be necessary to restart it.

**Comment by Garfield96** [comment]
I've restarted Jenkins build.

**Comment by HobbyProgrammer** [comment]
> 
> 
> I've restarted Jenkins build.

Thanks! Unfortunately, the same issue occured again...

**Pull request merged**